pkg/internal/vmcapabilities/vmcapabilities.go

```diff
@@ -0,0 +1,42 @@
+package vmcapabilities
+
+type Capabilities struct {
```

Comment:
Here's SKU cache from CAPZ which they use for the capability check https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/services/resourceskus/cache.go